src/Analyser/MutatingScope.php

```diff
@@ -2316,7 +2316,7 @@ public function resolveTypeByName(Name $name): TypeWithClassName
 
 		$originalClass = $this->resolveName($name);
 		if ($this->isInClass()) {
-			if ($this->inClosureBindScopeClass !== null && $this->inClosureBindScopeClass !== 'static' && $originalClass === $this->getClassReflection()->getName()) {
+			if ($this->inClosureBindScopeClass === $originalClass && $this->inClosureBindScopeClass !== 'static') {
```

Comment:
hmm it looks like tests are still green with the `'static'` handling removed. But I don't fully understand why and if this is covered by tests..

Reply:
ok, I could verify that even if `'static'` is used that this property is correctly resolved in the scope. Maybe via `resolveName`, not sure, but removing it should be fine then